tiledb/sm/query/query.cc

```diff
@@ -2149,6 +2153,11 @@ void Query::reset_coords_markers() {
 }
 
 void Query::copy_aggregates_data_to_user_buffer() {
```

Comment:
has_aggregates is added and only used in one place. I don't think we'll need that function anywhere else and this if statement is bound to go away soon. I would suggest to remove has_aggregates and use !default_channel_aggregates_.empty() here.